### ISSUE TITLE
Enhancement 

### ISSUE THREAD

**Issue by samkhanjar** [Fenugreek/comprehend#2]
Hi Deepak, I had a look at your library this morning and it's one of the best library I have used so far. I have few suggestions for enhancement for the library. Is it possible to plot graph of loss and cost rather than printing the values and also a graph classifying the data. For instance I have un-labelled dataset which I need to show on a graph. Is it possible to add this functionality in.

I'll be looking for your feedback...

Thanks in advance,

Sam

**Comment by Fenugreek** [comment]
Hi Sam,
    
    thanks for the compliment.
    
    For plotting the loss/cost, the way to do it is probably using
    Tensorboard, which this module will have to add support for. I am
    going to put it on my to-do list, but it's going to take a while. If
    you would like to help, you can try making the changes yourself --
    or point me to some other implementation of Tensordboard in some
    tensorflow project that I can then look into incorporating something
    similar here. My priority right now is to first upgrade to
    Tensorflow 1.0 and test this module under that.
    
    For a "graph classifying the data", I'm not sure what you mean, and
    I'm not sure whether it fits within the scope of this project. It
    seems to be a means of visualizing the predictions; something like
    Matplotlib might be the place to look for that. Here we are mostly
    concerned with training the network and producing the associated 
    predictions.
    
    Cheers,
    Deepak
    
    On 2/24/2017 8:33 PM, samkhanjar wrote:
    
    
      Hi Deepak, I had a look at your library this morning and it's
        one of the best library I have used so far. I have few
        suggestions for enhancement for the library. Is it possible to
        plot graph of loss and cost rather than printing the values and
        also a graph classifying the data. For instance I have
        un-labelled dataset which I need to show on a graph. Is it
        possible to add this functionality in.
      I'll be looking for your feedback...
      Thanks in advance,
      Sam
      —
        You are receiving this because you are subscribed to this
        thread.
        Reply to this email directly, view
          it on GitHub, or mute
          the thread.
      
        
          
          
        
        
      
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/Fenugreek/comprehend","title":"Fenugreek/comprehend","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/Fenugreek/comprehend"}},"updates":{"snippets":[{"icon":"DESCRIPTION","message":"Enhancement  (#2)"}],"action":{"name":"View Issue","url":"https://github.com/Fenugreek/comprehend/issues/2"}}}

**Comment by samkhanjar** [comment]
Hi Deepak how are you? Thank you again for your reply earlier. How would you add hidden layers before training? And also does your library do any validations? What i meant by validation is how accurate it predicts the output.

**Comment by Fenugreek** [comment]
Hi Sam,
you can add layers programmatically using the `layers` module which has the `layers.Layers` object that takes a list of regular `networks.*` objects as an argument to its constructor. If you prefer to work from the commandline, the `--add` option in `run.py` script can add one layer at a time to an object loaded from disk via the `--fromfile` option.

As for validation, the `run.py` training script splits given data into training and validation set (ratio controlled via `--validation` option), and prints costs on the validation set. If you'd like to check costs on a separate test set of data, after training the network, load it from disk and give it the test set data using the different filename for the `--data` option and make it compute average loss and cost via the `--loss` option.

I have now upgraded this library to Tensorflow 1.0, and am not working on your requested enhancement "plot graph of loss and cost" as that's best handled via Tensorboard, and I do not know what you mean by "graph classifying the data". So closing.

Thanks,
Deepak
